pkg/kubelet/volume/secret/secret.go

```diff
@@ -97,7 +97,7 @@ func (sv *secretVolume) SetUp() error {
 		return fmt.Errorf("Cannot setup secret volume %v because kube client is not configured", sv)
 	}
 
-	secret, err := kubeClient.Secrets(sv.secretRef.Namespace).Get(sv.secretRef.Name)
+	secret, err := kubeClient.Secrets(sv.podRef.Namespace).Get(sv.secretRef.Name)
```

Comment:
The problem with SecretRef (and likewise object references I am using for Persistent Volumes) is that it requires either a) the user to set the namespace on their pod's volume or b) setting the namespace on the ref in a default func.

The default func worked for me in my case, but @smarterclayton preferred adding to the NewBuilder func's arguments as opposed to a default func.